<a href="https://colab.research.google.com/github/muaaz-tahir/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/muaaz-tahir/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I am picking Lane 2: Refresh / Content Opportunity Scoring.

My question: which pages should FlyRank's content team review first for a
refresh, expansion, or pruning decision, given that they can only review a
limited number of pages each cycle?

I am picking this lane because it produces something a real person can act
on immediately: a ranked list, not just an observation. It also builds
naturally on the kind of pattern-checking I already did in Week 1 with this
same dataset.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Unit of analysis: one row = one content page (content_id).

Decision this improves: which pages the content review team looks at first,
out of everything in the inventory, given limited review capacity.

Action someone takes: a reviewer opens the top-ranked pages and decides
whether to refresh, expand, protect, prune, or simply monitor each one.

Cost of a wrong recommendation: if a page is ranked high but did not
actually need attention, the team loses reviewer time on a low-value page,
and a page that did need attention gets looked at later than it should.
This is a low-to-moderate cost mistake, since no page is changed
automatically. A human always makes the final call.

Why data or ML can help at all: with tens of thousands of pages and limited
review capacity, someone has to decide an order. A simple fixed rule (like
"page is old and still gets traffic") is a reasonable start, but it misses
pages where several weaker signals combine into real risk. A model can
combine signals like age, traffic trend, position, and word count into one
ranked list, which a fixed rule alone cannot do as well.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
!git clone https://github.com/muaaz-tahir/flyrank-ml-internship.git
import pandas as pd

df = pd.read_csv("flyrank-ml-internship/data/raw/content_refresh_anonymized.csv")

print(f"Total rows: {len(df)}")

# Number 1: how many pages are stale but still getting traffic
stale_visible = df[(df["days_since_last_update"] >= 180) & (df["impressions_90d"] >= 500)]
print(f"Stale but visible pages: {len(stale_visible)} ({len(stale_visible)/len(df)*100:.1f}%)")

# Number 2: how many pages are declining with real demand behind them
declining_with_demand = df[(df["trend_direction"] == "down") & (df["impressions_90d"] >= 100)]
print(f"Declining pages with demand: {len(declining_with_demand)} ({len(declining_with_demand)/len(df)*100:.1f}%)")

# Number 3: overall trend split
print(df["trend_direction"].value_counts(normalize=True) * 100)

fatal: destination path 'flyrank-ml-internship' already exists and is not an empty directory.
Total rows: 30000
Stale but visible pages: 17 (0.1%)
Declining pages with demand: 13152 (43.8%)
trend_direction
down      54.206667
stable    19.873333
up        14.626667
new        7.453333
flat       3.840000
Name: proportion, dtype: float64


My dataset has 30,000 pages.

13,152 pages (43.8%) are losing traffic but still get real search visits.
This is the biggest group. This is exactly what I want to find in this lane.

Only 17 pages (0.1%) are both old and still popular. This number is very
small. It means a simple rule like "old and popular" does not find much.
This shows why a smarter score or model is needed, not just one simple rule.

More than half of all pages (54.2%) are trending down. This shows the team
cannot review everything. A ranked list is genuinely needed here.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I CAN say: this project helps a person decide which page to check
first. It is based on real numbers from real pages.

What I CANNOT say: I cannot say that fixing a page will definitely bring
back its traffic. I would need a real test for that, not just this data.
I also cannot say anything about how Google's algorithm works. My numbers
only show what already happened, not why it happened.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.